functions.j2

```diff
@@ -52,13 +52,11 @@ function process_command_exit_status() {
   return "${exit_status}"
 }
 
-
-{# Loop through each provided command to be wrapped and make a single wrapper function for it -#}
-{% for command in commands -%}
-{%- if commands[command]["security"] is defined -%}
-function {{ command }}() {
+{# Macro to be used by the loops below #}
+{% macro function(function, tools, allow_filter, version_command, validations, security_tools) %}
+function {{ function }}() {
   arguments=("${@}")
-  easy_infra_{{ command }}_tools=({% for tool in commands[command]["security"] %}{{ tool }}{% if not loop.last %} {% endif %}{% endfor %})
+  easy_infra_{{ function | replace("-", "_") }}_tools=({% for tool in tools %}{{ tool | replace("-", "_") }}{% if not loop.last %} {% endif %}{% endfor %})
```

Comment:
So I decided to add these `| replace("-", "_")`s throughout instead of sanitizing prior to passing it in as something like `tool_sanitized`.  Open to push back on my approach

Reply:
I looked into it in a bit more detail and the various replacements happen in enough different situations that I don't think there is enough reuse to substantiate a change right now.  Any fix would be partial at best